attendance/views.py

```diff
@@ -17,6 +17,7 @@
 
 @group_required_multiple(["staff", "teacher"])
 def attendance(request):
+    user_group = str(request.user.groups.all().first())
```

Comment:
Resolved this instance by using request.user.groups.first() but raises a larger issue of handling a user in multiple groups in various parts of the application.  Should I create an issue for this?

Reply:
A lot of applications will handle this by using the same navigation code for all users and displaying all of the options that the user has access to, rather than having entirely separate flows for each user.

Reply:
I think I am somewhat doing this in "left_side_panel_attendance.html" where I check whether a user is a teacher or staff before displaying their side panel options.  But I'm not sure if that's what you mean.  I could simplify this application template code a whole lot by doing something like that for the base templates as well, however, I would still need the user_group variable in the context.  Although, again, not sure if this is what you mean.

Reply:
Instead of having `user_group` in context you'd have `user_groups` or equivalent in the context so you could test to see if that user had access to see a given menu option.

Something like:

```python
user_groups = set(user.groups.all().values_list('name', flat=True))
```

```html
{% if 'staff' in user_groups %}
  <!-- show staff options -->
{% endif %}
```

Although usually this kind of thing is often done at a somewhat higher level with a library like https://github.com/django-guardian/django-guardian

Reply:
I used context processors to provide all templates with the group that a user is in, and the classroom they are in (if any).  This eliminated a lot of redundancy in the templates, and will eliminate a lot of redundancy on the backend as well.  I will tackle the backend redundancy after this merge (I see a lot with registration right now and can find more I'm sure).